.github/scripts/deploy.sh

```diff
@@ -5,13 +5,59 @@ bucket=$1
 
 echo "current-version=$version" >> $GITHUB_OUTPUT
 
+delete_local_file() {
+    rm -f $versionsFileDir 
+}
+
+fileName="versions.csv"
+validVersionRegex="^[0-9]+\.[0-9]+\.[0-9]+$"
+
+# if file not found, returns NoSuchKey error
+aws s3api get-object --bucket $bucket --key "content/versions.csv" $fileName 
+
+versionsFileDir="./$fileName";
+currentVersions=$(<$versionsFileDir)
+echo CURRENT VERSIONS: $currentVersions
+
+allPairArr=(${currentVersions});
+
+# verify current versions in versions.csv file contain valid client,version pairs
+scriptString="arw-script"
+moduleString="arw-module"
+
+for singlePair in "${allPairArr[@]}"; 
+do 
+    IFS=',' read -ra singlePairArr <<< "$singlePair"
+
+    if [ ${singlePairArr[0]} != $scriptString ] && [ ${singlePairArr[0]} != $moduleString ]
+    then 
+        delete_local_file
+
+        echo ERROR: INVALID CLIENT
+        echo ${singlePairArr[0]}
+        exit 1
+    fi
+
+    if ! [[ ${singlePairArr[1]} =~ $validVersionRegex ]]
+    then
+        delete_local_file
+
+        
+        echo ERROR: INVALID VERSION
+        echo ${singlePairArr[1]}
+        exit 1
+    fi
+    
+done
+
```

Comment:
question: What are we trying to achieve here?

If the goal is to verify the file, I recommend doing so in a separate script, as a separate step, after the new file has been uploaded. The script would (1) download the from S3, (2) verify its correctness, and (3) fail the build.

Although such a step seems like overkill, so I'm OK with having it or not.

Reply:
Given we decided to remove the check from DP, I wanted to make sure we verify the file is formatted as expected during web client release. Will refactor to do in as separate script as a separate step.